driver-sync/src/test/functional/com/mongodb/client/AbstractUuidRepresentationTest.java

```diff
@@ -0,0 +1,321 @@
+            assertEquals(Binary.class, document.get("standard").getClass());
+            assertEquals(new Binary(BsonBinarySubType.UUID_STANDARD, standardEncodedValue), document.get("standard"));
+
+            assertEquals(Binary.class, document.get("legacy").getClass());
+            assertEquals(new Binary(BsonBinarySubType.UUID_LEGACY, standardEncodedValue), document.get("legacy"));
+        } else if (uuidRepresentation == UuidRepresentation.STANDARD) {
+            assertEquals(UUID.class, document.get("standard").getClass());
+            assertEquals(uuid, document.get("standard"));
+
+            assertEquals(Binary.class, document.get("legacy").getClass());
+            assertEquals(new Binary(BsonBinarySubType.UUID_LEGACY, standardEncodedValue), document.get("legacy"));
+        } else {
+            assertEquals(Binary.class, document.get("standard").getClass());
+            assertEquals(new Binary(BsonBinarySubType.UUID_STANDARD, standardEncodedValue), document.get("standard"));
+
+            assertEquals(UUID.class, document.get("legacy").getClass());
+            assertEquals(uuid, document.get("legacy"));
+        }
+    }
+
+    @Test
+    public void shouldDecodePojoWithStandardUuidRepresentation() {
+
+        bsonDocumentCollection.insertOne(new BsonDocument("_id", new BsonBinary(uuid, UuidRepresentation.STANDARD)));
+
+        try {
+            UuidIdPojo document = uuidIdPojoCollection.find().first();
+            assertNotNull(document);
+            assertEquals(uuid, document.getId());
+
+        } catch (BSONException e) {
+            assertNotEquals(UuidRepresentation.STANDARD, uuidRepresentation);
+        }
+    }
+
+    @Test
+    public void shouldDecodePojoWithLegacyUuidRepresentation() {
+
+        bsonDocumentCollection.insertOne(new BsonDocument("_id", new BsonBinary(uuid,
+                        uuidRepresentation == UuidRepresentation.UNSPECIFIED || uuidRepresentation == UuidRepresentation.STANDARD
+                                ? UuidRepresentation.PYTHON_LEGACY
+                                : uuidRepresentation)));
+
+        try {
+            UuidIdPojo document = uuidIdPojoCollection.find().first();
+            assertNotNull(document);
+            assertEquals(uuid, document.getId());
+        } catch (BSONException e) {
+            assertNotEquals(UuidRepresentation.JAVA_LEGACY, uuidRepresentation);
+            assertNotEquals(UuidRepresentation.C_SHARP_LEGACY, uuidRepresentation);
+            assertNotEquals(UuidRepresentation.PYTHON_LEGACY, uuidRepresentation);
+        }
+    }
+
+    @Parameterized.Parameters(name = "{0}-{1}")
+    public static Collection<Object[]> data() {
+        UUID uuid = UUID.fromString("00112233-4455-6677-8899-aabbccddeeff");
+
+        byte[] standardEncodedValue = Hex.decode("00112233445566778899AABBCCDDEEFF");
+
+        List<Object[]> data = new ArrayList<Object[]>();
+        data.add(new Object[]{
```

Comment:
What happens if `BsonBinarySubType.UUID_STANDARD` but  `UuidRepresentation.JAVA_LEGACY`? What happens in 3.11 and what will happen in 3.12

Reply:
In this patch it decodes to `Binary`, but it should decode to UUID until the 4.0 release.  Good catch.